src/ray/rpc/worker/direct_actor_client.h

```diff
@@ -124,6 +129,10 @@ class DirectActorClient : public std::enable_shared_from_this<DirectActorClient>
 
   /// The max sequence number we have processed responses for.
   int64_t max_finished_seq_no_ GUARDED_BY(mutex_) = -1;
+
+  /// The task id we are currently sending requests for. When this changes,
+  /// the max finished seq no counter is reset.
+  std::string cur_caller_id_;
```

Comment:
Was this a bug?

Also, shouldn't this be a TaskID?

Reply:
Yep. It's encoded as binary -- no need to decode.